packages/sirius-web/frontend/sirius-web-view-fork/src/contributions/ForkTreeItemContextMenuContribution.tsx

```diff
@@ -0,0 +1,146 @@
+  GQLCreateForkedStudioMutationData,
+  GQLErrorPayload,
+} from './ForkViewMenuAction.types';
+
+const forkViewMutation = gql`
+  mutation createForkedStudio($input: CreateForkedStudioInput!) {
+    createForkedStudio(input: $input) {
+      __typename
+      ... on CreateProjectSuccessPayload {
+        project {
+          id
+        }
+      }
+      ... on ErrorPayload {
+        message
+      }
+    }
+  }
+`;
+
+const isErrorPayload = (payload: CreateForkedStudioPayload): payload is GQLErrorPayload =>
+  payload.__typename === 'ErrorPayload';
+
+const isSuccessPayload = (payload: CreateForkedStudioPayload): payload is CreateProjectSuccessPayload =>
+  payload.__typename === 'CreateProjectSuccessPayload';
+
+export const ForkTreeItemContextMenuContribution = forwardRef(
+  ({ editingContextId, item }: TreeItemContextMenuComponentProps, ref: React.ForwardedRef<HTMLLIElement>) => {
+    const [createProject, { data, error }] = useMutation<GQLCreateForkedStudioMutationData>(forkViewMutation);
+    const { addErrorMessage } = useMultiToast();
+    const navigate = useNavigate();
+
+    const [state, setState] = useState<ForkViewMenuActionStates>({
+      isOpen: false,
+    });
+
+    const representationIds = item.kind.startsWith('siriusComponents://representation') ? [item.id] : [];
+    const { data: representationMetaData } = useRepresentationMetadata({ editingContextId, representationIds });
+
+    const handleClickOpen = () => {
+      setState((prevState) => ({ ...prevState, isOpen: true }));
+    };
+
+    const handleClose = () => {
+      setState((prevState) => ({ ...prevState, isOpen: false }));
+    };
+
+    const forkViewMutationHandler = () => {
+      const input: GQLCreateForkedStudioInput = {
+        id: crypto.randomUUID(),
+        editingContextId,
+        representationId: item.id,
+        tableId: '',
+      };
+      createProject({ variables: { input } });
+    };
+
+    if (data && isErrorPayload(data.createForkedStudio)) {
+      addErrorMessage(data.createForkedStudio.message);
+    }
+
+    if (error) {
+      addErrorMessage('An unexpected error has occurred, please refresh the page');
+    }
+
+    useEffect(() => {
+      if (data && isSuccessPayload(data.createForkedStudio)) {
+        navigate(`/projects/${data.createForkedStudio.project.id}/edit`);
+        navigate(0);
+      }
+    }, [data]);
+
+    if (!representationMetaData) {
+      return null;
+    } else {
+      if (representationMetaData.viewer.editingContext.representations.edges.find((edge) => !edge.node.isViewBased)) {
+        return null;
+      }
+    }
+
+    return (
+      <Fragment key="fork-tree-item-context-menu-contribution">
+        <MenuItem onClick={handleClickOpen} ref={ref} key="new-fork" data-testid="fork-action">
+          <ListItemIcon>
+            <EditNoteIcon fontSize="small" />
+          </ListItemIcon>
+          <ListItemText>Fork View Model</ListItemText>
+        </MenuItem>
+        <Dialog
```

Comment:
Do not keep dialog in the DOM like that, it can create sizable performance issue (this one would not but I prefer not to see this pattern)